### PR TITLE
qnetd-cluster-list: sort by node_id when add new client

### PR DESCRIPTION
**Pull request by liangxin1300**
Hi @jfriesse ,

I suggest to sort qnetd client by client id when listing them, otherwise when some nodes rebooting, list results will inconsistent:)

Thank you for review!

### CODE REVIEW

**Comment by jfriesse** [comment]
@knet-ci-bot ok to test

File: qdevices/qnetd-cluster-list.c [hunk 1]
```diff
@@ -61,10 +61,17 @@ qnetd_cluster_list_find_by_name(struct qnetd_cluster_list *list,
 	return (NULL);
 }
 
+static int
+client_node_id_cmp(const struct qnetd_client *entry1, const struct qnetd_client *entry2)
+{
+	return (entry1->node_id - entry2->node_id);
```
Comment (jfriesse) [inline qdevices/qnetd-cluster-list.c:67]: Please replace this with `if (entry1->node_id < entry2->node_id) { return (-1); } else if (...`. I know it is longer and this substraction should work, but in general it doesn't work and it is source of quite a big bugs (https://marc.info/?l=openbsd-tech&m=141806297022265&w=2) and it is just better to not have it in codebase.

File: qdevices/qnetd-cluster-list.c [hunk 2]
```diff
@@ -84,7 +91,19 @@ qnetd_cluster_list_add_client(struct qnetd_cluster_list *list, struct qnetd_clie
 		TAILQ_INSERT_TAIL(list, cluster, entries);
 	}
 
-	TAILQ_INSERT_TAIL(&cluster->client_list, client, cluster_entries);
+	entry = TAILQ_FIRST(&cluster->client_list);
```
Comment (jfriesse) [inline qdevices/qnetd-cluster-list.c:94]: Just please add comment there (sorting by nodeid because ...)

**Review by jfriesse — CHANGES_REQUESTED** [review]
Hi @liangxin1300 I like patch idea a lot (good catch there) and in general I like also the patch, but there are two small nitpicks. I think it is good to go when fixed.

File: qdevices/qnetd-cluster-list.c (hunk 1 above)
Comment (liangxin1300) [reply qdevices/qnetd-cluster-list.c]: Changed
Thanks!

File: qdevices/qnetd-cluster-list.c (hunk 2 above)
Comment (liangxin1300) [reply qdevices/qnetd-cluster-list.c:103]: Changed
Thanks!

**Comment by liangxin1300** [comment]
> Hi @liangxin1300 I like patch idea a lot (good catch there) and in general I like also the patch, but there are two small nitpicks. I think it is good to go when fixed.

Changed and also synced to needle branch

**Pull request closed without merging**